command/ca/acme/eab/list.go

```diff
@@ -0,0 +1,178 @@
+package eab
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/ca"
+	"github.com/smallstep/cli/flags"
+	"github.com/smallstep/cli/utils/cautils"
+	"github.com/urfave/cli"
+	"go.step.sm/cli-utils/errs"
+)
+
+func listCommand() cli.Command {
+	return cli.Command{
+		Name:   "list",
+		Action: cli.ActionFunc(listAction),
+		Usage:  "list all ACME External Account Binding Keys",
+		UsageText: `**step beta ca acme eab list** <provisioner> [<reference>]
+[**--limit**=<number>] [**--admin-cert**=<file>] [**--admin-key**=<file>]
+[**--admin-provisioner**=<string>] [**--admin-subject**=<string>]
+[**--password-file**=<file>] [**--ca-url**=<uri>] [**--root**=<file>]
+[**--context**=<name>]`,
+		Flags: []cli.Flag{
+			flags.Limit,
+			flags.NoPager,
+			flags.AdminCert,
+			flags.AdminKey,
+			flags.AdminProvisioner,
+			flags.AdminSubject,
+			flags.PasswordFile,
+			flags.CaURL,
+			flags.Root,
+			flags.Context,
+		},
+		Description: `**step beta ca acme eab list** lists all ACME External Account Binding (EAB) Keys.
+
+Output will go to stdout by default. If many EAB keys are stored in the ACME provisioner, output will be sent to $PAGER (when set). 
+
+## POSITIONAL ARGUMENTS
+
+<provisioner>
+: Name of the provisioner to list ACME EAB keys for
+
+<reference>
+: (Optional) reference (from external system) for the key to be listed
+
+
+## EXAMPLES
+
+List all ACME External Account Binding Keys:
+'''
+$ step beta ca acme eab list my_acme_provisioner
+'''
+
+Show ACME External Account Binding Key with specific reference:
+'''
+$ step beta ca acme eab list my_acme_provisioner my_reference
+'''
+`,
+	}
+}
+
+func listAction(ctx *cli.Context) (err error) {
+	if err := errs.MinMaxNumberOfArguments(ctx, 1, 2); err != nil {
+		return err
+	}
+
+	args := ctx.Args()
+	provisioner := args.Get(0)
+
+	reference := ""
+	if ctx.NArg() == 2 {
+		reference = args.Get(1)
+	}
+
+	client, err := cautils.NewAdminClient(ctx)
+	if err != nil {
+		return errors.Wrap(err, "error creating admin client")
+	}
+
+	var out io.WriteCloser
+	var cmd *exec.Cmd
+
+	usePager := true
+	if ctx.IsSet("no-pager") {
+		usePager = !ctx.Bool("no-pager")
+	}
+
+	// the pipeSignalHandler goroutine ensures that the parent process is closed
+	// whenever one of its childs is killed.
+	go pipeSignalHandler()
+
+	// prepare the $PAGER command to run when not disabled and when available
+	pager := os.Getenv("PAGER")
+	if usePager && pager != "" {
+		cmd = exec.Command(pager)
+		var err error
+		out, err = cmd.StdinPipe()
+		if err != nil {
+			return errors.Wrap(err, "error setting stdin")
+		}
+		defer out.Close()
+		cmd.Stdout = os.Stdout
+	} else {
+		out = os.Stdout
+	}
+
+	// default to API paging per 100 entities
+	limit := uint(0)
+	if ctx.IsSet("limit") {
+		limit = ctx.Uint("limit")
+	}
+
+	cursor := ""
+	format := "%-36s%-28s%-16s%-30s%-30s%-36s%s\n"
+	firstIteration := true
+	startedPager := false
+
+	for {
```

Comment:
I like the way you did this iteration. Gonna use this style elsewhere.

Reply:
First time implementing `$PAGER` and I like it too! There may be rough edges with how various terminals / $PAGERs work. I've only tested it with `less`. `--no-pager` probably unblocks those cases. Got that one from `git`. Also was inspired by @tashian's (a.o.) https://clig.dev/ 😃. 

I thought about making this a bit more generic, so that it could be in `cli-utils` sometime, but I'd need to do some more concrete instances first to see what works best. 

@maraino suggested performing the API calls in the same loop as printing (as opposed to loading all results first and then printing), which results in now only performing API calls when the user actively pages down. Simple, but effective! 🙂